### DISCUSSION TITLE
How to execute an action on value change

### DISCUSSION THREAD

**Discussion by kkharji** [emilk/egui#1501] [Q&A] [answered]
Hello, so I have a setting page that has theme option. I'd like to execute an action if the value got changed. I could store the value in local variable and check if it's changed but I wonder if there is an api for that.

My code

```rust
ComboBox::from_id_source("theme")
                .width(125.)
                .selected_text(self.state.mode.as_ref())
                .show_ui(ui, |ui| {
                    for option in Mode::iter().collect::<Vec<_>>() {
                        ui.selectable_value(&mut self.state.mode, option, option.as_ref());

                    }
                });
```

**Comment by coderedart** [comment, marked as answer]
you can simply check that the value in `self.state.mode` has changed before and after running the UI code.
also `ui.selectable_value()` returns a `Response` which has a `.changed()` method which will return true if the value changed.

**Reply by kkharji** [reply]
😍 thanks @coderedart 

```rust

Mode::iter().for_each(|o| {
    ui.selectable_value(&mut self.state.mode, o, o.as_ref())
        .changed()
        .then(|| self.configure_styles(ui.ctx(), false));
});
```
